src/databroker_elasticsearch/elasticcallback.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+"""\
+Callback for adding data to elastic search from run engine.
+"""
+
+from bluesky.callbacks.core import CallbackBase
+from databroker_elasticsearch.elasticindex import ElasticIndex
+
+
+class ElasticCallback(CallbackBase):
+    """
+    Callback for inserting start metadata into an `Elasticsearch` instance.
+
+    This wraps `ElasticIndex` adapter for adding translated documents to
+    a given Elasticsearch index.
+
+    Attributes
+    ----------
+    esindex : ElasticIndex
+        The ElasticIndex object for converting documents and inserting
+        them to Elasticsearch.
+    """
+
+
+    def __init__(self, esindex: ElasticIndex):
+        self.esindex = esindex
+        return
+
+
+    @classmethod
+    def from_config(cls, config):
+        """
+        Create a new ElasticCallback instance using a configuration dictionary.
+
+        Parameters
+        ----------
+        config : dict
+            The configuration dictionary that describes ElasticIndex.
+            It must contain "databroker-elasticsearch" key.
+
+        Returns
+        -------
+        ElasticCallback
+        """
+        from databroker_elasticsearch.elasticindex import ElasticIndex
+        esindex = ElasticIndex.from_config(config)
+        rv = cls(esindex)
+        return rv
+
+
+    def rebuild(self, headers, purge=False):
+        """Re-export start documents in given headers to Elasticsearch index.
+
+        Parameters
+        ----------
+        headers : iterable
+            The sequence of objects with "start" attribute which is exported.
+            This is usually an iterable of `databroker.Header` objects.
+        purge : bool, optional
```

Comment:
What is the benefit of providing this to users?

Reply:
The document map will probably be unstable for some time.
This lets beamline scientists redo the ES when the map changes.
It also let's them reindex some documents (say from the last week) if the callback was inactive.